src/main/java/Deadline.java

```diff
@@ -0,0 +1,25 @@
+public class Deadline extends Task{
+    private String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    private Deadline(Task t, String by) {
+        super(t);
+        this.by = by;
+    }
+
+    public Deadline markDone() {
+        return new Deadline(super.markDone(), this.by);
+    }
```

Comment:
Should `markDone`, `markUndone` and `toString` have `@Override` annotation? Also applicable to `Todo` and`Event`

```suggestion
    @Override
    public Deadline markDone() {
        return new Deadline(super.markDone(), this.by);
    }
```